### PR TITLE
[Merged by Bors] - feat: port Data.Multiset.Lattice

### PR DESCRIPTION
**Pull request by 4e554c4c**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Multiset/Lattice.lean [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+namespace Multiset
+
+variable {α : Type _}
+
+/-! ### sup -/
+
+
+section Sup
+
+-- can be defined with just `[has_bot α]` where some lemmas hold without requiring `[order_bot α]`
```
Comment (jcommelin) [inline Mathlib/Data/Multiset/Lattice.lean:29]:
```suggestion
-- can be defined with just `[Bot α]` where some lemmas hold without requiring `[OrderBot α]`
```

File: Mathlib/Data/Multiset/Lattice.lean [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+#align multiset.nodup_sup_iff Multiset.nodup_sup_iff
+
+end Sup
+
+/-! ### inf -/
+
+
+section Inf
+
+-- can be defined with just `[has_top α]` where some lemmas hold without requiring `[order_top α]`
```
Comment (jcommelin) [inline Mathlib/Data/Multiset/Lattice.lean:113]:
```suggestion
-- can be defined with just `[Top α]` where some lemmas hold without requiring `[OrderTop α]`
```

File: Mathlib/Data/Multiset/Lattice.lean [hunk 3]
```diff
@@ -10,6 +10,7 @@ Authors: Mario Carneiro
 -/
 import Mathlib.Data.Multiset.FinsetOps
 import Mathlib.Data.Multiset.Fold
+import Mathlib.Data.Multiset.Basic
```
Comment (jcommelin) [inline Mathlib/Data/Multiset/Lattice.lean:13]: Why do you need this import?

Comment (4e554c4c) [reply Mathlib/Data/Multiset/Lattice.lean:13]: I thought it was causing the "porting note" issue in l98, but that turned out to be a different problem
removing :)

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**